src/SignalR/clients/csharp/Client.Core/src/HubConnectionExtensions.OnResult.cs

```diff
@@ -0,0 +1,486 @@
+    /// </summary>
+    /// <typeparam name="T1">The first argument type.</typeparam>
+    /// <typeparam name="T2">The second argument type.</typeparam>
+    /// <typeparam name="T3">The third argument type.</typeparam>
+    /// <typeparam name="T4">The fourth argument type.</typeparam>
+    /// <typeparam name="T5">The fifth argument type.</typeparam>
+    /// <typeparam name="T6">The sixth argument type.</typeparam>
+    /// <typeparam name="T7">The seventh argument type.</typeparam>
+    /// <typeparam name="TResult">The return type the handler returns.</typeparam>
+    /// <param name="hubConnection">The hub connection.</param>
+    /// <param name="methodName">The name of the hub method to define.</param>
+    /// <param name="handler">The handler that will be raised when the hub method is invoked.</param>
+    /// <returns>A subscription that can be disposed to unsubscribe from the hub method.</returns>
+    public static IDisposable On<T1, T2, T3, T4, T5, T6, T7, TResult>(this HubConnection hubConnection, string methodName, Func<T1, T2, T3, T4, T5, T6, T7, Task<TResult>> handler)
+    {
+        if (hubConnection == null)
+        {
+            throw new ArgumentNullException(nameof(hubConnection));
+        }
+
+        return hubConnection.On(methodName,
+            new[] { typeof(T1), typeof(T2), typeof(T3), typeof(T4), typeof(T5), typeof(T6), typeof(T7) },
+            args => handler((T1)args[0]!, (T2)args[1]!, (T3)args[2]!, (T4)args[3]!, (T5)args[4]!, (T6)args[5]!, (T7)args[6]!));
+    }
+
+    /// <summary>
+    /// Registers a handler that will be invoked when the hub method with the specified method name is invoked.
+    /// Returns value returned by handler to server if the server requests a result.
+    /// </summary>
+    /// <typeparam name="T1">The first argument type.</typeparam>
+    /// <typeparam name="T2">The second argument type.</typeparam>
+    /// <typeparam name="T3">The third argument type.</typeparam>
+    /// <typeparam name="T4">The fourth argument type.</typeparam>
+    /// <typeparam name="T5">The fifth argument type.</typeparam>
+    /// <typeparam name="T6">The sixth argument type.</typeparam>
+    /// <typeparam name="T7">The seventh argument type.</typeparam>
+    /// <typeparam name="T8">The eighth argument type.</typeparam>
+    /// <typeparam name="TResult">The return type the handler returns.</typeparam>
+    /// <param name="hubConnection">The hub connection.</param>
+    /// <param name="methodName">The name of the hub method to define.</param>
+    /// <param name="handler">The handler that will be raised when the hub method is invoked.</param>
+    /// <returns>A subscription that can be disposed to unsubscribe from the hub method.</returns>
+    public static IDisposable On<T1, T2, T3, T4, T5, T6, T7, T8, TResult>(this HubConnection hubConnection, string methodName, Func<T1, T2, T3, T4, T5, T6, T7, T8, Task<TResult>> handler)
```

Comment:
8? Not 10?

Reply:
Just copied what we had for the non-result On methods.